c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```
@@ -5451,6 +5474,437 @@ bool constructor(JSContext *cx, unsigned argc, Value *vp) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain ? character");
+    return false;
+  }
+  if (strchr(key_chars, '*') != NULL) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain * character");
+    return false;
+  }
+  if (strchr(key_chars, '[') != NULL) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain [ character");
+    return false;
+  }
+  if (strchr(key_chars, ']') != NULL) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain ] character");
+    return false;
+  }
+  if (strchr(key_chars, '\n') != NULL) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain newline character");
+    return false;
+  }
+  if (strchr(key_chars, '\r') != NULL) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not contain carriage return character");
+    return false;
+  }
+  auto acme_challenge = ".well-known/acme-challenge/";
+  if (strncmp(key_chars, acme_challenge, strlen(acme_challenge)) == 0) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not start with .well-known/acme-challenge/");
+    return false;
+  }
+
+  if (strcmp(key_chars, ".") == 0) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not be \".\"");
+    return false;
+  }
+  if (strcmp(key_chars, "..") == 0) {
+    JS_ReportErrorASCII(cx, "ObjectStore key can not be \"..\"");
+    return false;
+  }
+
+  HandleValue body_val = args.get(1);
+
+  // We currently support five types of body inputs:
+  // - byte sequence
+  // - buffer source
+  // - USV strings
+  // - URLSearchParams
+  // - ReadableStream
+  // After the other other options are checked explicitly, all other inputs are
+  // encoded to a UTF8 string to be treated as a USV string.
+  // TODO: Support the other possible inputs to Body.
+
+  RootedObject body_obj(cx, body_val.isObject() ? &body_val.toObject() : nullptr);
+
+  if (body_obj && JS::IsReadableStream(body_obj)) {
+    JS_ReportErrorLatin1(cx, "Streaming into an ObjectStore is not yet supported");
```

After speaking with @acfoltzer - we think that the hostcalls do support streaming from a BodyHandle which came from the host such as a client request's BodyHandle. Any BodyHandle that came from the guest can not yet be streamed into ObjectStore such as a ReadableStream or a TransformStream.

---

There's also an extra challenge here is we wanted it. ObjectStore entries can only be a max of 30MB. We could count the size of the stream as we pass it through to ObjectStore and if it exceeds 30MB we could throw a JS Error which states "ObjectStore value can not be more than 30 Megabytes in size"